### PR TITLE
ButtonSecondary to TS

### LINKED ISSUE #18888: Migrate components to TS: ButtonSecondary
### Description
Migrate `ButtonSecondary` component to TypeScript

Dependencies
- https://github.com/MetaMask/metamask-extension/issues/19239

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- Fixes #18888 

### Before

https://github.com/MetaMask/metamask-extension/assets/79097544/bd27a4ea-730a-4988-88d1-ea2cb0669dd5

### After

https://github.com/MetaMask/metamask-extension/assets/79097544/5863fc09-741f-4a61-b2be-af40034d1c7b

image

## Manual Testing Steps

- Check controls for `ButtonSecondary` on storybook
- `yarn jest ui/components/component-library/button-secondary`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/button-secondary/button-secondary.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  ) => {
+    const buttonColor = danger ? Color.errorDefault : Color.primaryDefault;
+    return (
+      <ButtonBase
+        backgroundColor={Color.transparent}
+        borderColor={buttonColor}
+        color={buttonColor}
+        className={classnames(className, 'mm-button-secondary', {
+          'mm-button-secondary--type-danger': danger || false,
+          'mm-button-secondary--disabled': disabled || false,
```
Comment (dhruvv173) [inline ui/components/component-library/button-secondary/button-secondary.tsx:32]: wasnt marked as required in the js file hence this

File: ui/components/component-library/button-secondary/button-secondary.types.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import type { ButtonBaseStyleUtilityProps } from '../button-base/button-base.types';
+import type { PolymorphicComponentPropWithRef } from '../box';
+
+export enum ButtonSecondarySize {
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+}
+
+export type ValidButtonTagType = 'button' | 'a';
```
[the remark this answers is not in the document]
Comment (dhruvv173) [reply ui/components/component-library/button-secondary/button-secondary.types.ts]: done [`here`](https://github.com/MetaMask/metamask-extension/pull/20411/commits/43f722e5316681ce791bc107b7df9b0657725957)

**Review by garrettbear — APPROVED** [review]
LGTM

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Thanks for your contribution @dhruvv173! Pulled branch and checked
- Storybook stories and controls
- Storybook docs
- Searched for `BUTTON_SECONDARY_SIZES` to check no instances in codebase
- No errors in console

**Pull request merged**